File: src/main/java/com/github/eendroroy/sdk/openweather/response/SysResponse.java
package com.github.eendroroy.sdk.openweather.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author indrajit
 */

@ToString
@Setter
@Getter
@JsonRootName("sys")
public class SysResponse implements BaseResponse {
    @JsonProperty("type")
    public Integer type;

    @JsonProperty("id")
    public Integer id;

    @JsonProperty("message")
    public Double message;

    @JsonProperty("country")
    public String country;

    @JsonProperty("sunrise")
    public Integer sunrise;

    @JsonProperty("sunset")
    public Integer sunset;}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/response/SysResponse.kt
package com.github.eendroroy.sdk.openweathermap.response

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonRootName

/**
 * @author indrajit
 */
@JsonRootName("sys")
class SysResponse : BaseResponse {
    @JsonProperty("type")
    var type: Int? = null

    @JsonProperty("id")
    var id: Int? = null

    @JsonProperty("message")
    var message: Double? = null

    @JsonProperty("country")
    var country: String? = null

    @JsonProperty("sunrise")
    var sunrise: Int? = null

    @JsonProperty("sunset")
    var sunset: Int? = null
}